app/livechat/server/lib/Livechat.js

```diff
@@ -1200,6 +1200,10 @@ export const Livechat = {
 		};
 		LivechatVisitors.updateById(contactId, updateUser);
 	},
+	updateCallStatus(callId, rid, status) {
+		Rooms.setCallStatus(rid, status);
```

Comment:
I didn't understand why we need to maintain the same status on both `room` and `message` object. Doesn't it seem redundant?

Reply:
or is there any difference between the statuses stored in each object? If so could you please explain?

Reply:
Here I followed the same way Jitsi is done. Instead of `callStatus` we have `jitsiTimeout` stored in both the `room` and `message` object to know the status of the call. If the timeout expires then it means the call ended otherwise the call hasn't yet ended. 
We need to make sure that one call is initiated at a time in one room. In order to know whether the room has an ongoing call or not we need the call status in the `room` object. And to maintain the call status initiated by the message action buttons, we need the call status in the `message` object.

Reply:
> We need to make sure that one call is initiated at a time in one room. In order to know whether the room has an ongoing call or not we need the call status in the room object.

:top: This totally makes sense. I also think that the room object should be the right place to store the call status since we have a rule that only one call can be initiated at a time in one room.

> And to maintain the call status initiated by the message action buttons, we need the call status in the message object.

:top: This doesn't makes sense, why can't we check the room's `callStatus`  property when any action button is clicked? All the checks you're performing [here](https://github.com/RocketChat/Rocket.Chat/pull/22689/files#diff-0e697798268df9d4a2d39482864eb4cf479141030a7287dd1208d51ff8c07885R12) and [here](https://github.com/RocketChat/Rocket.Chat/pull/22689/files#diff-0e697798268df9d4a2d39482864eb4cf479141030a7287dd1208d51ff8c07885R23) on the message `callStatus` property, why not perform the same check on the room `callStatus` property?